lib/puppetlabs_spec_helper/rake_tasks.rb

```diff
@@ -141,6 +142,15 @@ def clone_repo(scm, remote, target, ref=nil, branch=nil, flags = nil)
     fail "Unfortunately #{scm} is not supported yet"
   end
   result = system("#{scm} #{args.flatten.join ' '}")
+  unless dirpath.nil?
+    tmpdir = SecureRandom.hex
+    system('setopt -s dotglob')
+    system("mkdir #{target}/#{tmpdir}")
```

Comment:
Please use http://ruby-doc.org/stdlib-2.3.1/libdoc/fileutils/rdoc/FileUtils.html instead of these calls. It makes it much more secure, efficient and readable.